client/my-sites/backup/backup-date-picker/test/index.jsx

```diff
@@ -55,206 +59,184 @@ describe( 'BackupDatePicker', () => {
-		} ) );
-
-	test( 'Navigates backward when the previous date is focused and the spacebar is pressed', () =>
-		new Promise( ( done ) => {
-			const selectedDate = moment( '2020-01-01' );
-			const onDateChange = ( date ) => {
-				expect( date.diff( selectedDate, 'days' ) ).toEqual( -1 );
-				done();
-			};
-
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } />
-			);
-
-			const button = picker.find( '.backup-date-picker__select-date--previous' );
-			button.simulate( 'keydown', { key: ' ' } );
-		} ) );
-
-	test( 'Navigates forward when the next date is focused and the spacebar is pressed', () =>
-		new Promise( ( done ) => {
-			const selectedDate = moment( '2020-01-01' );
-			const onDateChange = ( date ) => {
-				expect( date.diff( selectedDate, 'days' ) ).toEqual( 1 );
-				done();
-			};
-
-			const picker = shallow(
-				<BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } />
-			);
-
-			const button = picker.find( '.backup-date-picker__select-date--next' );
-			button.simulate( 'keydown', { key: ' ' } );
-		} ) );
+	test( 'Navigates backward when the previous date is clicked', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Dec 31, 2019' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.subtract( 1, 'day' ) );
+	} );
+
+	test( 'Navigates forward when the next date is clicked', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getByText( 'Jan 2, 2020' );
+		await user.click( button );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.add( 1, 'day' ) );
+	} );
+
+	test( 'Navigates backward when the previous date is focused and the spacebar is pressed', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getAllByRole( 'button' )[ 0 ];
+		button.focus();
+		await user.keyboard( '[Space]' );
+
+		expect( onDateChange ).toHaveBeenCalledWith( selectedDate.subtract( 1, 'day' ) );
+	} );
+
+	test( 'Navigates forward when the next date is focused and the spacebar is pressed', async () => {
+		const user = userEvent.setup();
+
+		const selectedDate = moment( '2020-01-01' );
+		const onDateChange = jest.fn();
+
+		render( <BackupDatePicker selectedDate={ selectedDate } onDateChange={ onDateChange } /> );
+
+		const button = screen.getAllByRole( 'button' )[ 2 ];
```

Comment:
Noting that if we don't label buttons and the only way to locate one is to use the index points to an accessibility issue. Not something we necessarily have to fix here, as it existed before, but something I wanted to point out.

Reply:
@tyxla Sounds right. I have added labels for those buttons, hope it looks good now